usaspending_api/awards/management/commands/loadcontracts.py

```diff
@@ -228,34 +227,92 @@ def fetch_country_code(vendor_country_code):
 
 
-def get_or_create_location(row):
-    country_code = fetch_country_code(row['vendorcountrycode'])
+'''
+PLACE OF PERFORMANCE
+
+CITY: placeofperformancecity
+CD: placeofperformancecongressionaldistrict
+CD: pop_cd
+COUNTRY: placeofperformancecountrycode
+ZIPCODE: placeofperformancezipcode
+STATE: pop_state_code
+STATE: statecode
+
+VENDOR
+
+CITY: city
+CD: congressionaldistrict
+CD: vendor_cd
+COUNTRY: vendorcountrycode
+ZIPCODE: zipcode
+STATE: vendor_state_code
+STATE: state
+VENDORSTREET1: streetaddress
+VENDORSTREET2: streetaddress2
+VENDORSTREET3: streetaddress3
+
+countryoforigin ????? blank in most cases anyway
+locationcode ????? blank in most cases
+'''
+
+
+def get_or_create_location(row, mode="vendor"):
+    place_of_performance_mapping = {
+        "city": row["placeofperformancecity"],
+        "congressionaldistrict": row["placeofperformancecongressionaldistrict"][2:],  # Need to strip the state off the front
+        "country": row["placeofperformancecountrycode"],
+        "zipcode": row["placeofperformancezipcode"].replace("-", ""),  # Either ZIP5, or ZIP5+4, sometimes with hypens
+        "state_code": row["pop_state_code"].split(":")[0]  # Format is VA: VIRGINIA, so we need to grab the first bit
+    }
+
+    vendor_mapping = {
+        "city": row["city"],
+        "congressionaldistrict": row["vendor_cd"].zfill(2),  # Need to add leading zeroes here
+        "country": row["vendorcountrycode"],  # Never actually a country code, just the string name
+        "zipcode": row["zipcode"].replace("-", ""),
+        "state_code": row["vendor_state_code"].split(":")[0],
+        "street1": row["streetaddress"],
+        "street2": row["streetaddress2"],
+        "street3": row["streetaddress3"]
+    }
+
+    mapping = vendor_mapping
+
+    if mode == "place_of_performance":
+        mapping = place_of_performance_mapping
+
+    country_code = fetch_country_code(mapping["country"])
     location_dict = {
         "location_country_code": country_code,
-        "location_address_line1": row["streetaddress"],
-        "location_address_line2": row["streetaddress2"],
-        "location_address_line3": row["streetaddress3"],
     }
 
     if country_code.country_code == "USA":
         location_dict.update(
-            location_zip5=row["zipcode"][:5],
-            location_zip_last4=row["zipcode"].replace("-", "")[5:],
-            location_state_code=row["state"],
-            location_city_name=row["city"],
+            location_zip5=mapping["zipcode"][:5],
+            location_zip_last4=mapping["zipcode"][5:],
+            location_state_code=mapping["state_code"],
+            location_city_name=mapping["city"],
+            location_congressional_code=mapping["congressionaldistrict"]
         )
     else:
         location_dict.update(
-            location_foreign_postal_code=row["zipcode"],
-            location_foreign_province=row["state"],
+            location_foreign_postal_code=mapping["zipcode"],
```

Comment:
Some of this logic (e.g., setting foreign postal code to zip code when appropriate) also applies to the submission ETL. At what point does it make sense to centralize some of it to ensure that things remain consistent regardless of data source? I suspect we're already out of sync on some things.